src/Client.cpp

```diff
@@ -568,12 +568,14 @@ bool CClient::PutClient(const CMessage& Message) {
     Msg.SetClient(this);
     Msg.SetNetwork(m_pNetwork);
 
-    bool bReturn = false;
-    NETWORKMODULECALL(OnSendToClientMessage(Msg), m_pUser, m_pNetwork, this,
-                      &bReturn);
-    if (bReturn) return false;
+    bool bReturn = PutClientRaw(Msg.ToString());
```

Comment:
You destroyed the ability of modules to block sending some things to client

Reply:
Agreed: I think like the real fix is creating a new callback
Sorry I should have said that (and perhaps have my change in a single clean commit.)
I don't really know the process or who looks and when so cheers for taking a peek.

But now I want to give a whirl to the vanilla ZNC too (no clientbuffer.)  From the issue it sounds like it isn't fixed..  

But if the buffer is updated based on response from the client, then it should have always worked ?

I'll look at the code.